ledger/src/blockstore_processor.rs

```diff
@@ -853,17 +853,17 @@ pub fn process_blockstore_from_root(
     // We might be promptly restarted after bad capitalization was detected while creating newer snapshot.
     // In that case, we're most likely restored from the last good snapshot and replayed up to this root.
     // So again check here for the bad capitalization to avoid to continue until the next snapshot creation.
-    if !bank_forks
-        .read()
-        .unwrap()
-        .root_bank()
-        .calculate_and_verify_capitalization(debug_verify)
-    {
-        return Err(
-            BlockstoreProcessorError::RootBankWithMismatchedCapitalization(
-                bank_forks.read().unwrap().root(),
-            ),
-        );
+
+    let bank = bank_forks.read().unwrap().root_bank();
+    if start_slot != bank.slot() {
+        bank.wait_for_startup_verification_complete();
+        if !bank.calculate_and_verify_capitalization(debug_verify) {
```

Comment:
with this change the only thing changing is that we wait for the existing bg hash calc to complete if it was run in the bg. But we aren't skipping a redundant startup account verification, right? So I suggest you update the pr title.

Reply:
We should skip it if the root slot is unchanged after `load_frozen_forks`, which should be the case, for example, when bootstrapping without an existing snapshot

Reply:
oh. doh. yep.